src/sbt-test/sbt-bundle/basic/build.sbt

```diff
@@ -15,6 +15,22 @@ BundleKeys.roles := Set("web-server")
 
 BundleKeys.endpoints += "other" -> Endpoint("http", 0, Set(URI("http://:9001/simple-test")))
 BundleKeys.endpoints += "akka-remote" -> Endpoint("tcp")
+BundleKeys.endpoints += "extras" -> Endpoint("http", 0, "ping-service",
+  Acl(
+    HttpRequestMappings(
+      HttpPath("/bar-1"),
+      HttpPath("/bar-2", "GET"),
+      HttpPath("/bar-3", "GET", "/other-bar-3"),
+      HttpPathBeg("/beg-1"),
+      HttpPathBeg("/beg-2", "GET"),
+      HttpPathBeg("/beg-3", "GET", "/other-beg-3"),
+      HttpPathRegex("/regex1/[a|b|c]"),
+      HttpPathRegex("/regex2/[a|b|c]", "GET"),
+      HttpPathRegex("/regex3/[a|b|c]", "GET", "/other-regex-3")
+    )),
+  Acl(TcpRequestMappings(9001, 9002)),
+  Acl(UdpRequestMappings(19001, 19002))
+)
```

Comment:
I've been playing around with this. How about something like this:

```scala
BundleKeys.endpoints= "extras" -> Endpoint(
  "http", 
  "ping-service",  // Note make the bindPort the third param so we don't have to name the service
  acls = Set(
    Http(
	  "/bar-1",
	  "GET" -> "/bar-2",
	  "GET" -> "/bar-3" -> "/other-bar-3",
	  "^/beg-1".r, // A regex with a ^ at the beginning, but no $ at the end means "path begin"
      "GET" -> "^/beg-2".r,
      "GET" -> "^/beg-3".r -> "/other-beg-3",
      "^/regex1/[a|b|c]$".r, // A regex with a ^ at the beginning, and a $ at the end is a "path regex"
      "GET" -> "/regex2/[a|b|c]".r, // No ^ and no $ also means "path regex" (more convenient)
      "GET" -> "/regex3/[a|b|c]".r -> "/other-regex-3"
	  ),
	Tcp(9001,  9002),
	Udp(19001, 19002)
  )
```

The above is all made quite type safe given:

```scala
  import scala.util.matching.Regex 
    
  sealed abstract class RequestFamily
  
  object Http {
	object Method extends Enumeration {
	  val GET, POST, PUT, DELETE, HEAD, TRACE, CONNECT = Value
	}
	
	implicit def method(m: String): Method.Value =
	  Method.withName(m)
	  
    case class Request(method: Option[Method.Value], path: Either[String, Regex], rewrite: Option[String])
	
	implicit def request1(r: String): Request = 
	  Request(None, Left(r), None)
  	implicit def regexRequest1(r: Regex): Request = 
  	  Request(None, Right(r), None)

  	implicit def request2(r: (String, String)): Request = 
  	  Request(Some(r._1), Left(r._2), None)
  	implicit def regexRequest2(r: (String, Regex)): Request = 
  	  Request(Some(r._1), Right(r._2), None)

  	implicit def request3(r: ((String, String), String)): Request = 
  	  Request(Some(r._1._1), Left(r._1._2), Some(r._2))
	implicit def regexRequest3(r: ((String, Regex), String)): Request = 
	  Request(Some(r._1._1), Right(r._1._2), Some(r._2))
  }
  case class Http(requests: Http.Request*) extends RequestFamily
  
  object Tcp {
    case class Request(port: Int)
	
	implicit def request1(p: Int): Request = 
	  Request(p)
  }
  case class Tcp(requests: Tcp.Request*) extends RequestFamily
  
  object Udp {
    case class Request(port: Int)
	
	implicit def request1(p: Int): Request = 
	  Request(p)
  }
  case class Udp(requests: Udp.Request*) extends RequestFamily
```

  ...and acls is a `Set[RequestFamily]`.

WDYT?